lib/Doctrine/ODM/PHPCR/UnitOfWork.php

```diff
@@ -813,7 +800,7 @@ private function doScheduleInsert($document, &$visited, $overrideIdGenerator = n
                 // TODO: Change Tracking Deferred Explicit
                 break;
             case self::STATE_REMOVED:
-                unset($this->scheduledRemovals[$oid]);
+                $this->treeOpQueue->unqueue(TreeOperation::OP_REMOVE, $oid);
```

Comment:
if you want to record operations, i think you would need to queue another add operation instead. i am sure it would be possible to find some hairbrained scenario where this makes a difference. and probably some very real usecase with listeners doing many things could actually result in the same scenario and actually make sense...